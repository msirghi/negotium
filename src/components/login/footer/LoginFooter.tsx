import { makeStyles } from '@mui/styles';
import colors from '../../../common/styles/colors';

const useStyles = makeStyles({
  link: {
    cursor: 'pointer',
    color: colors.primaries.lightBlue_1,
  },
  container: {
    fontSize: 13,
    textAlign: 'center'
  },
});

export const LoginFooter = () => {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      Donâ€™t have an account? <span className={classes.link}>Sign Up</span>
    </div>
  );
};
