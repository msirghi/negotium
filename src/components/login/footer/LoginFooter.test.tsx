import renderer, { act } from 'react-test-renderer';
import { LoginFooter } from './LoginFooter';
import { mount } from 'enzyme';

const mockPush = jest.fn();

jest.mock('next/router', () => ({
  useRouter: () => ({
    route: 'inbox',
    query: {},
    push: mockPush,
  }),
}));

describe('Login footer', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<LoginFooter />);
    expect(tree).toMatchSnapshot();
  });

  it('should handle sign in click', () => {
    const wrapper = mount(<LoginFooter />);
    act(() => {
      wrapper.find('span').simulate('click');
    });
    expect(mockPush).toBeCalled();
  });
});
