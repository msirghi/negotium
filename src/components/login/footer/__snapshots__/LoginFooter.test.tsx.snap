// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login footer should match snapshot 1`] = `
<div
  className="makeStyles-container-2"
>
  login.dontHaveAnAccount
   
  <span
    className="makeStyles-link-1"
    onClick={[Function]}
  >
    login.signUp
  </span>
</div>
`;
